app/main.py

```diff
@@ -1,6 +1,35 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cache_storage = dict()
+
+    def add_to_storage(func_name: str, args: tuple) -> int:
+
+        func_result = func(*args)
+        launch_in = str(args)
+        cache_storage[func_name][launch_in] = func_result
+        print("Calculating new result")
+        return func_result
+
+    @wraps(func)
+    def wrapper(*args) -> Callable:
+
+        func_name = func.__name__
+        func_result = None
```

Comment:
The variable `func_result` is initialized with `None` but is immediately reassigned in all branches of the code. Consider removing this initialization to clean up the code.